File: Assets/_scripts/fish/animation/FishYawingAnimation.js
class FishYawingAnimation extends FishAnimation
{
    var yawDrift : float = 5;
    var yawPeriod : float = 5;
    var sideDrift : float = 0.2;

    private var phase : float = 0;
    private var period : float = Mathf.Infinity;

    function LateUpdate(){
        updatePeriodAndPhase();

        // Debug.Log(String.Format("velocity: {0}, period: {1}", rigidbody.velocity.magnitude, period));    

        yawShift = yawDrift * Mathf.Sin(phase);    
        transform.localEulerAngles += Vector3(0, yawShift, 0);

        sideShift = -sideDrift * Mathf.Cos(phase - Mathf.PI / 4);
        transform.localPosition += Vector3(sideShift, 0, 0);
    }

    //////    //////    //////    //////    //////

    function updatePeriodAndPhase(){
        var fishLength = 5;
        period = yawPeriod/transform.parent.rigidbody.velocity.magnitude;        
        phase +=  2 * Mathf.PI * (Time.deltaTime / period);    
    }

    function isIdle() {
        return period > 10;
    }
}

File: Assets/_scripts/fish/animation/FishYawingAnimation.cs
using UnityEngine;
using System.Collections;

public class FishYawingAnimation : FishAnimation 
{
    public float yawDrift  = 5f;
    public float yawPeriod  = 5f;
    public float sideDrift  = 0.2f;

    private float phase  = 0;
    private float period  = Mathf.Infinity;

    void LateUpdate(){
        Profiler.StartProfile(PT.Yawing);

        updatePeriodAndPhase();

        // Debug.Log(String.Format("velocity: {0}, period: {1}", rigidbody.velocity.magnitude, period));    

        float yawShift = yawDrift * Mathf.Sin(phase);    
        transform.localEulerAngles += new Vector3(0, yawShift, 0);

        float sideShift = -sideDrift * Mathf.Cos(phase - Mathf.PI / 4);
        transform.localPosition += new Vector3(sideShift, 0, 0);

        Profiler.EndProfile(PT.Yawing);
    }

    //////    //////    //////    //////    //////

    void updatePeriodAndPhase(){
        period = yawPeriod/transform.parent.rigidbody.velocity.magnitude;        
        phase +=  2 * Mathf.PI * (Time.deltaTime / period);    
    }

    bool isIdle() {
        return period > 10;
    }
}